src/libraries/System.Console/src/System/Console.cs

```diff
@@ -235,16 +235,23 @@ static TextWriter EnsureInitialized()
 
         private static TextWriter CreateOutputWriter(Stream outputStream)
         {
-            return outputStream == Stream.Null ?
-                TextWriter.Null :
-                TextWriter.Synchronized(new StreamWriter(
-                    stream: outputStream,
-                    encoding: OutputEncoding.RemovePreamble(), // This ensures no prefix is written to the stream.
-                    bufferSize: WriteBufferSize,
-                    leaveOpen: true)
-                    {
-                        AutoFlush = true
-                    });
+            if (outputStream == Stream.Null)
+                return TextWriter.Null;
+
+            StreamWriter writer = new StreamWriter(
+                stream: outputStream,
+                encoding: OutputEncoding.RemovePreamble(), // This ensures no prefix is written to the stream.
+                bufferSize: WriteBufferSize,
+                leaveOpen: true
+            ) {
+                AutoFlush = true
+            };
+
+#if TARGET_BROWSER && !FEATURE_WASM_MANAGED_THREADS
+            return writer;
+#else
+            return TextWriter.Synchronized(writer);
```

Comment:
Rather than all of these changes, would it make more sense to just change TextWriter.Synchronized itself for wasm?

Reply:
Isn't that a public API? Changing what it returns could break user code that relies on what it returns. vs just changing whether we wrap streams internally, which is an internal implementation detail (though I can imagine user code relying on that too.)

Reply:
The API is typed to return the base type. We do not consider it to be a breaking change to return a different more derived instance than was previously being returned. In this case that derived type isn't even public.

Reply:
Verified that changing Synchronized works and is much simpler, thanks!